File: helpers/objects/contrived.py
from functools import total_ordering


@total_ordering
class Contrived:

    def __init__(self, letter, number):
        self.letter = letter
        self.number = number

    def __str__(self):
        return f"Contrived ({self.letter}:{self.number})"

    def __lt__(self, other):
        if not isinstance(other, self.__class__):
            return NotImplemented
        if self.letter == other.letter:
            return self.number < other.number

    def __eq__(self, other):
        if not isinstance(other, self.__class__):
            return NotImplemented
        return self.__dict__ == other.__dict__

File: helpers/objects/java/Contrived.java
package helpers.objects.java;

import java.util.Objects;

/**
 * Class used in searching algorithms to show how collections
 * of custom objects can also be searched.
 */
public final class Contrived implements Comparable<Contrived> {
    private final int number;
    private final char letter;

    public Contrived(int number, char letter) {
        this.number = number;
        this.letter = letter;
    }

    private final int sum() {
        return this.number + this.letter;
    }

    @Override
    public boolean equals(Object other) {
        if (this == other) {
            return true;
        }

        if (other == null || this.getClass() != other.getClass())  {
            return false;
        }

        Contrived contrived = (Contrived) other;

        return number == contrived.number && letter == contrived.letter;
    }

    @Override
    public int hashCode() {
        return Objects.hash(number, letter);
    }

    @Override
    public int compareTo(Contrived other) {
        if (this.sum() == other.sum()) {
            return 0;
        } else if (this.sum() > other.sum()) {
            return 1;
        }

        return -1;
    }

    @Override
    public String toString() {
        return "Contrived: " + this.number + ":" + this.letter;
    }
}